jembi/JeMPI
# Dockerizing the bootstrapper app

**Ticket:** JMPI-771
**Other Related Tickets:** N/A
**Other Related PRS:**  N/A

----

## Describe of changes

This change dockerizes the current bootstrapper, allowing it to be part of the deployed stack/ a service in the swarm. This allows it to have access to the backend services deployed in the swarm for data management, primarily kafka, postgres, and dgraph. 

**Please note**

- the dockerized version assumes that you have started all other hub services i.e you have run `devops/linux/docker/helper/scripts/d-stack-up-hub-containers.sh` 
- The is a wrapper script that can call the dockerized bootstrapper (see `devops/linux/docker/helper/bootstrapper/bootstrapper-docker.sh`). You can use this to call the bootstrapper methods, for example ` ./helper/bootstrapper/bootstrapper-docker.sh data resetAll`


## How to test / configure

- Run JeMPI as you would normally 
- After the hub services start you should see logging similar to

```
2023-12-11 00:27:07 [ INFO ] Jempi Bootstrapper > Data:  Resetting Postgres data and schemas.
2023-12-11 00:27:07 [ INFO ] Jempi Bootstrapper > Data:  Deleting Postgres data
2023-12-11 00:27:07 [ INFO ] Jempi Bootstrapper > Data:  Deleting Postgres tables
2023-12-11 00:27:07 [ INFO ] Jempi Bootstrapper > Data:  Loading Postgres schema data.
2023-12-11 00:27:07 [ INFO ] org.jembi.jempi.libmpi.dgraph.LibDgraph:  LibDgraph Constructor
2023-12-11 00:27:07 [ INFO ] org.jembi.jempi.libmpi.dgraph.LibDgraph:  [alpha-01] [9080]
2023-12-11 00:27:07 [ INFO ] org.jembi.jempi.libmpi.dgraph.LibDgraph:  LibDgraph Constructor
2023-12-11 00:27:07 [ INFO ] org.jembi.jempi.libmpi.dgraph.LibDgraph:  [alpha-01] [9080]
2023-12-11 00:27:07 [ INFO ] Jempi Bootstrapper > Data > DGraph:  Resetting DGraph data and schemas.
2023-12-11 00:27:07 [ INFO ] Jempi Bootstrapper > Data > DGraph:  Deleting DGraph data and schemas.
2023-12-11 00:27:08 [ INFO ] Jempi Bootstrapper > Data > DGraph:  Loading DGraph schema data.
2023-12-11 00:27:09 [ INFO ] Jempi Bootstrapper > Data > Kafka:  Connecting to the kafka bootstrap server 'kafka-01:9092'
2023-12-11 00:27:09 [ INFO ] Jempi Bootstrapper > Data > Kafka:  Connecting to the kafka bootstrap server 'kafka-01:9092'
2023-12-11 00:27:09 [ INFO ] Jempi Bootstrapper > Data > Kafka:  Resetting kafka data and schemas.
2023-12-11 00:27:09 [ INFO ] Jempi Bootstrapper > Data > Kafka:  Deleting kafka topics.
```
